test/razorTests/Microsoft.AspNetCore.Razor.VSCode.Grammar.Test/tests/typeparamDirective.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { describe, it } from '@jest/globals';
+import { assertMatchesSnapshot } from './infrastructure/testUtilities';
+
+// See GrammarTests.test.ts for details on exporting this test suite instead of running in place.
+
+export function RunTypeparamDirectiveSuite() {
+    describe('@typeparam directive', () => {
+        it('No type', async () => {
+            await assertMatchesSnapshot('@typeparam');
+        });
+
+        it('No type spaced', async () => {
+            await assertMatchesSnapshot('@typeparam              ');
+        });
+
+        it('Incomplete type, generic', async () => {
+            await assertMatchesSnapshot('@typeparam SomeViewBase<string');
```

Comment:
Great job, thinking to test generics with this! Maybe for completeness, a test that includes the closing `>` would be good?

Reply:
Thanks for the tip! Although I can't take the credit for that since I merely modeled the test cases after the other test cases in the test suite.